### PR TITLE
docs(cdaas): RBAC Manual Approval 

### PR DESCRIPTION
**Pull request by aimeeu**
Add concept, task; update deploy ref

Resolves Jira: [CDAAS-1808]
Relates to: [CDAAS-1736]  code ticket where rules are defined

Deploy previews:
Concept: https://deploy-preview-1679--armory-docs.netlify.app/cd-as-a-service/concepts/deployment/role-based-manual-approval/
Task: https://deploy-preview-1679--armory-docs.netlify.app/cd-as-a-service/tasks/deploy/role-based-manual-approval/

[CDAAS-1808]: https://armory.atlassian.net/browse/CDAAS-1808?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[CDAAS-1736]: https://armory.atlassian.net/browse/CDAAS-1736?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: content/en/cd-as-a-service/reference/deployfile/ref-deployment-file.md [hunk 1]
```diff
@@ -605,14 +607,16 @@ redirectTrafficAfter:
 
 When you configure a manual judgment, the deployment waits for manual approval through the UI. You can either approve the deployment or roll the deployment back if something doesn't look right. Do not provide a `duration` or `unit` value when defining a judgment-based pause.
 
-`strategies.<strategyName>.blueGreen.redirectTrafficAfter.pause.untilApproved: true`
+- `strategies.<strategyName>.blueGreen.redirectTrafficAfter.pause.untilApproved: true`
+- `targets.<targetName>.constraints.beforeDeployment.pause.requiresRole` (Optional) list of RBAC roles that can issue a manual approval
```
Comment (parth-srivastav) [inline content/en/cd-as-a-service/reference/deployfile/ref-deployment-file.md]:
```suggestion
- `strategies.<strategyName>.blueGreen.redirectTrafficAfter.pause.requiresRole` (Optional) list of RBAC roles that can issue a manual approval
```

File: content/en/cd-as-a-service/reference/deployfile/ref-deployment-file.md [hunk 2]
```diff
@@ -391,7 +391,8 @@ steps:
 
 When you configure a manual judgment, the deployment waits when it hits the corresponding weight threshold. At that point, you can either approve the deployment so far and let it continue or roll the deployment back if something doesn't look right.
 
-`strategies.<strategyName>.canary.steps.pause.untilApproved: true`
+- `strategies.<strategyName>.canary.steps.pause.untilApproved: true`
+- `targets.<targetName>.constraints.beforeDeployment.pause.requiresRole` (Optional) list of RBAC roles that can issue a manual approval
```
Comment (parth-srivastav) [inline content/en/cd-as-a-service/reference/deployfile/ref-deployment-file.md]:
```suggestion
- `strategies.<strategyName>.canary.steps.pause.requiresRole` (Optional) list of RBAC roles that can issue a manual approval
```

File: content/en/cd-as-a-service/reference/deployfile/ref-deployment-file.md [hunk 3]
```diff
@@ -206,9 +205,10 @@ targets:
       beforeDeployment:
```
Comment (parth-srivastav) [inline content/en/cd-as-a-service/reference/deployfile/ref-deployment-file.md:205]: Unrelated to this PR: we are missing the spec for `afterDeployment` constraints

Comment (aimeeu) [reply content/en/cd-as-a-service/reference/deployfile/ref-deployment-file.md:205]: Agree. The entire ref-deployment-file should be double-checked. I know I haven't been keeping it updated based on new fields. Big ooops.

**Pull request merged**